protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpServer.java

```diff
@@ -77,75 +75,62 @@ public void initChannel(Channel channel) {
             bootstrap.option(ChannelOption.TCP_FASTOPEN, 3);
         }
 
-        ChannelFuture future = bootstrap.bind();
-
-        if (wait) {
-            try {
-                future.sync();
-            } catch (InterruptedException e) {
+        CompletableFuture<Void> handleFuture = new CompletableFuture<>();
+        bootstrap.bind().addListener((ChannelFutureListener) future1 -> {
+            if (future1.isSuccess()) {
+                channel = future1.channel();
+                if (callback != null) {
+                    callback.run();
+                }
+            } else {
+                log.error("Failed to bind connection listener.", future1.cause());
```

Comment:
```suggestion
        bootstrap.bind().addListener((ChannelFutureListener) future -> {
            if (future.isSuccess()) {
                channel = future.channel();
                if (callback != null) {
                    callback.run();
                }
            } else {
                log.error("Failed to bind connection listener.", future.cause());
```
There's no other variable called future; so this works too (and looks a bit cleaner)